src/Microsoft.AspNetCore.Authentication/RemoteAuthenticationResult.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Claims;
+
+namespace Microsoft.AspNetCore.Authentication
+{
+    /// <summary>
+    /// Contains the result of an Authenticate call
+    /// </summary>
+    public class RemoteAuthenticationResult : AuthenticateResult
```

Comment:
This concept probably belongs in HttpAbstractions, should align it with HandleRequest since that's the concept there.

I think this should probably be the return type for https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNetCore.Authentication.Abstractions/IAuthenticationRequestHandler.cs#L18